apps/qa/src/pages/ingest/ingest.py

```diff
@@ -2,20 +2,41 @@ def ingest():
     import streamlit as st
     from .helpers import archive_raw_data
     from pathlib import Path
+    import time
 
     st.title("Ingest Dataset")
 
-    dataset_name = st.text_input("Dataset Name")
-    version = st.text_input("Version")
-    uploaded_file = st.file_uploader("Choose a file")
+    if "running" not in st.session_state:
+        st.session_state.running = False
+
+    dataset_name = st.text_input("Dataset Name", disabled=st.session_state.running)
+    version = st.text_input("Version", disabled=st.session_state.running)
+    uploaded_file = st.file_uploader("Choose a file", disabled=st.session_state.running)
     s3_path = Path("inbox") / dataset_name / version
 
-    ingest_button_pressed = st.button("Ingest")
+    disabled = st.session_state.get("running", False)
+    ingest_button_pressed = st.button("Ingest", disabled=st.session_state.running)
     if ingest_button_pressed == True:
         if dataset_name and version and uploaded_file:
-            file_name = uploaded_file.name
-            file_path = archive_raw_data(
-                dataset_name, version, uploaded_file, file_name
-            )
+            st.session_state.running = True
+            st.rerun()
```

Comment:
So I don't think we should need to use `rerun`. I'm guessing you drew inspiration from the GRU page, but that's an unusual case in that it needs to track status of external orchestration - it's refreshing to see what the status of actions are on github. We _should_ (I think) be able to get by without any `sleep` or `rerun`. At least for this initial part. In general in the streamlit model, when a widget is interacted with, the screen automatically reloads. So the button press is all we need.

For our button, we might want to use an `on_click` function rather than just treating it as a bool - this way, we can set state, and this state can be used at the beginning to determine what we show.

Let's also get more specific with state - one, let's use notation `key["value"]` rather than `key.value` just because that's generally the way we've gone in our codebase. But let's have a `st.session_state["ingest"]` as a dict in which we can store these, to make sure this page wouldn't interfere with another page.